domain/application/service/unit.go

```diff
@@ -226,6 +231,40 @@ func (s *Service) UpdateCAASUnit(ctx context.Context, unitName coreunit.Name, pa
 	return nil
 }
 
+// SetReportedUnitAgentVersion sets the reported agent version for the
+// supplied unit name. Reported agent version is the version that the agent
+// binary on this unit has reported it is running.
+//
+// The following errors are possible:
+// - [coreerrors.NotValid] if the reportedVersion is not valid.
+// - [coreerrors.NotSupported] if the architecture is not supported.
+// - [applicationerrors.UnitNotFound] - when the unit does not exist.
+func (s *Service) SetReportedUnitAgentVersion(ctx context.Context, unitName coreunit.Name, reportedVersion agentbinary.Version) error {
+	if err := reportedVersion.Validate(); err != nil {
+		return errors.Errorf("reported agent version %v is not valid: %w", reportedVersion, err)
+	}
+
+	unitUUID, err := s.st.GetUnitUUIDByName(ctx, unitName)
```

Comment:
It's not enough just to get the unit UUID - we want to ensure the unit is not dead. See the existing mongo code.
Use the `GetUnitLife` method.
```
func (u *Unit) SetAgentVersion(v version.Binary) (err error) {
	defer errors.DeferredAnnotatef(&err, "setting agent version for unit %q", u)
	if err = checkVersionValidity(v); err != nil {
		return err
	}
	versionedTool := &tools.Tools{Version: v}
	ops := []txn.Op{{
		C:      unitsC,
		Id:     u.doc.DocID,
		Assert: notDeadDoc,
		Update: bson.D{{"$set", bson.D{{"tools", versionedTool}}}},
	}}
	if err := u.st.db().RunTransaction(ops); err != nil {
		return onAbort(err, stateerrors.ErrDead)
	}
	u.doc.Tools = versionedTool
	return nil
}
```

Reply:
To clarify my comment, as I think it isn't clear, the intent of the GetUnitLife method is needed. We can discuss f2f